test/pytest/service-desktop/test_security.py

```diff
@@ -1,33 +1,73 @@
 # Copyright (c) 2017-2021, Mudita Sp. z.o.o. All rights reserved.
 # For licensing, see https://github.com/mudita/MuditaOS/LICENSE.md
+import logging
 
 import pytest
 import time
 
 from harness.interface.defs import status
-
+from harness.api.developermode import PhoneModeLock
+from harness.api.security import GetPhoneLockStatus, GetPhoneLockTime, SetPhoneLockOff
+from harness.request import TransactionError
+
+
+class SecurityTester:
+    def __init__(self, harness):
+        self.harness = harness
+
+    def assert_phone_unlocked(self):
+        assert GetPhoneLockStatus().run(self.harness).response.status == status["NoContent"]
+
+    def assert_phone_locked(self):
+        try:
+            GetPhoneLockStatus().run(self.harness).response.status
+        except TransactionError as error:
+            assert error.status == status["Forbidden"]
+        else:
+            logging.error("Phone is unlocked, but should be locked!")
+            assert False
+
+    def assert_passcode_format_validation(self, passcode: list):
+        try:
+            SetPhoneLockOff(passcode).run(self.harness)
+        except TransactionError as error:
+            assert error.status == status["BadRequest"]
+        else:
+            logging.error("Passcode with incorrect format was accepted!")
+            assert False
+
+    def assert_no_time_lock(self):
+        try:
+            GetPhoneLockTime().run(self.harness)
+        except TransactionError as error:
+            assert error.status == status["UnprocessableEntity"]
+        else:
+            logging.error("Phone is time locked unexpectedly!")
+            assert False
+
+    def unlock_phone(self, passcode: list = None):
+        assert SetPhoneLockOff(passcode).run(self.harness).response.status == status["NoContent"]
+        time.sleep(0.5)
```

Comment:
why is this sleep required?

Reply:
On a physical device unlocking the phone takes (a short) time.
Without any sleep in this place, this operation could not be finished in the OS, before check after it.
0.1 or even less would be enough probably, but a bit more is reserved just in case.
Probably it could be combined with await and _assert_phone_locked_, but not sure if it's needed.
This solution is simple and reliable enough. The state of the lock is always checked after this operation, and if sleeping for 0.5 is not enough, this means that something is bad.
tl;dr
ihmo its reliable enough for this test

Reply:
we have a lot of tests like that and in my opinion its a bad practice.
1. we are waiting for something which is a side-effect but user doesn't know what it is - we should wait for event explicitly
2. it can be slightly delayed and it will be issue, when it should be. Or it would be frequently used, but will happen much faster - but code will be super slow as there would be lots of sleep code in it.

changing this to something we could trust should be fairly easy. Unfortunatelly this is quite common pattern in our tests. I could agree to ie. helping you out changing it to something better (few hour top)?

Reply:
Will be changed in the separate task: https://appnroll.atlassian.net/browse/CP-953